bin/inx-chronicle/src/stardust_inx/mod.rs

```diff
@@ -192,3 +193,36 @@ impl HandleEvent<Report<Syncer>> for InxWorker {
         Ok(())
     }
 }
+
+#[async_trait]
+impl HandleEvent<IsHealthy> for InxWorker {
+    async fn handle_event(
+        &mut self,
+        _cx: &mut ActorContext<Self>,
+        IsHealthy(sender): IsHealthy,
+        inx_client: &mut Self::State,
+    ) -> Result<(), Self::Error> {
+        let node_status = NodeStatus::try_from(inx_client.read_node_status(NoParams {}).await?.into_inner())
+            .map_err(InxError::InxTypeConversion)?;
+
+        let mut healthy = true;
+        healthy &= node_status.is_healthy;
+
+        let latest_inserted_ms = self.db.get_latest_milestone().await?;
+        healthy &= latest_inserted_ms.map_or(false, |latest_inserted_ms| {
+            node_status.confirmed_milestone.milestone_info.milestone_index == latest_inserted_ms
+        });
```

Comment:
Shouldn't this be a range, rather than a concrete comparison? What would happen if the node is currently still processing the cone of the milestone in `node_status`?

Reply:
We should check for `last_inserted_ms >= confirmed_milestone_index - 3` (or some other magical number). Just remember that the milestone index can also be 0 or 1.

Reply:
What's the actual issue with returning unhealthy once in a blue moon due to this?

Reply:
If it's easily avoidable, we should do it. We don't know how people will use this endpoint, or what kind of infra they build on top of it.

Probably it's fine to check against the current and the previous milestone though.

Reply:
Makes sense to me